scipy/sparse/_compressed.py

```diff
@@ -55,6 +55,8 @@ def __init__(self, arg1, shape=None, dtype=None, copy=False):
                     coo = self._coo_container(arg1, shape=shape, dtype=dtype)
                     arrays = coo._coo_to_compressed(self._swap)
                     self.indptr, self.indices, self.data, self._shape = arrays
+                    if not self.has_canonical_format:
```

Comment:
I'm not sure it makes sense to check this here. We just assigned the member arrays, so we can't know if we're already in canonical format at this point.

Reply:
But I think we have to call ` _sparsetools.csr_has_canonical_format` somehow, and if I understand the `has_canonical_format` property, it calls sparsetools when the private property `_has_canonical_format` is not yet defined.  Is there a better way to check canonical format?

Reply:
CJ showed me that the `sum_duplicates` sparsetools actually checks whether the sparse array has canonical format -- so we are just repeating work if we check manually before summing. So, I've removed those calls.